linkcheck/strformat.py

```diff
@@ -69,7 +71,7 @@ def ascii_safe (s):
     @return: encoded ASCII version of s, or None if s was None
     @rtype: string
     """
-    if isinstance(s, unicode):
+    if isinstance(s, str_text):
         s = s.encode('ascii', 'ignore')
     return s
```

Comment:
I only just noticed all those `@rtype: string` annotations that should maybe be mass-replaced with `@rtype: bytes` where it makes sense?  (In a separate PR maybe, this one's already large enough.)

(I'm not 100% convinced it makes sense for ascii_safe() to return bytes on Python 3, but then I haven't actually looked at where it's used.)

Reply:
I'm tending to think we park this in the new PR and revisit after going Python 3 only.